ujt/app.py

```diff
@@ -116,109 +150,191 @@ def generate_graph_elements(node_name_message_map: Dict[str, Node],
         A list of dictionary objects, each containing information regarding a single node (Service or Client) or edge (Dependency).
     """
 
-    compute_status.compute_node_statuses(node_name_message_map,
-                                         client_name_message_map)
+    return (
+        converters.cytoscape_elements_from_nodes(node_name_message_map) +
+        converters.cytoscape_elements_from_clients(client_name_message_map))
+
+
+def generate_dropdown_options(client_name_message_map: Dict[str, Client]):
+    return [
+        {
+            "label": name,
+            "value": name,
+        } for name in client_name_message_map.keys()
+    ]
+
+
+@app.callback(
+    [
+        Output("cytoscape-graph",
+               "elements"),
+        Output("client-dropdown",
+               "options")
+    ],
+    [Input("refresh-button",
+           "n_clicks_timestamp")],
+)
+def refresh(n_clicks_timestamp):
+    cache.clear()
+    node_name_message_map, client_name_message_map = read_local_data()
 
-    return (converters.cytoscape_elements_from_nodes(node_name_message_map) +
-            converters.cytoscape_elements_from_clients(client_name_message_map))
+    cytoscape_graph_elements = generate_graph_elements(
+        node_name_message_map,
+        client_name_message_map,
+    )
+
+    client_dropdown_options = generate_dropdown_options(client_name_message_map)
+
+    return cytoscape_graph_elements, client_dropdown_options
+
+
+@app.callback(
+    Output("node-info-panel",
+           "children"),
+    [Input("cytoscape-graph",
+           "tapNode")],
+)
+def generate_node_info_panel(tap_node):
```

Comment:
Here's the new code to generate the node info panel including up to 3 Datatables holding data describing the Node's SLIs, Dependencies, and Children